### ISSUE TITLE
Autocomplete of backticks doesn't work when using keyboard locale with dead keys

### ISSUE THREAD

**Issue by esseb** [atom/bracket-matcher#205]
Autocompleting of backticks was added in https://github.com/atom/bracket-matcher/issues/32, but when using a keyboard locale with dead keys it doesn't work as expected.

Steps to reproduce:
1. Switch to a keyboard locale with dead keys support (I use "Norwegian Bokmål - Norwegian Extended" on Mac)
2. Select a word in Atom
3. Type a backtick (I need to press &lt;shift&gt; and the key left of backspace)

Result:
The word selected in step 2 is replaced by a single backtick, rendered with an underline to indicate it's an active dead key. If I now press the keyboard button "e" I end up with the character "è".

Further info:
If I press &lt;cmd&gt;+&lt;z&gt; for undo once after step 3 I get the expected result, so it looks like the backtick autocomplete operation succeeded before the dead key got inserted.

I don't know if it's possible to prevent the OS level dead key feature, but if it is, it should be prevented when typing a backtick while a word is selected, and only then. Ideally this would be a configurable option set to `off` by default because I can imagine the current behaviour is desired by many users.

**Comment by esseb** [comment]
Tested using Atom v1.3.3

Perhaps a custom package is the way to solve this issue. I'd still need a way to somehow prevent the OS level dead key feature, unless I'm somehow able to trigger and undo action.

**Comment by lee-dohm** [comment]
This sounds like it is related to atom/atom-keymap#37.

**Comment by esseb** [comment]
I don't think so. The shortcut does work, as evidenced by pressing cmd+z, it's just that the dead key behaviour triggers in addition afterwards.

**Comment by lee-dohm** [comment]
Hmmmmm ok. I'll see if anyone on the team has experience with key mappings for dead keys.

**Comment by abe33** [comment]
Behaviour of dead keys is always a bit tricky, but as far as I can tell it works properly.

Let me explain: 

When pressing the backtick once, the keystroke isn't complete, that the concept behind dead keys, you'll need to press another key for the dead key to be resolved. 
Using the space key is generally the way to get the dead character only, pressing any other key will attempt to resolve the dead key, and may ultimately add the character if it cannot be resolved
When using backtick and then space you get the expected behaviour, everything else will trigger a dead key resolution.

Here's a little gif that demonstrate how it behaves with space: 

![backtick-autocomplete](https://cloud.githubusercontent.com/assets/247140/12273740/d0d1d35e-b967-11e5-90df-b3dd7f5d5d8d.gif)

In the first sequence: 

1. I select the word
2. I press the backtick key, the word is replaced temporarily since I may overwrite it if the dead key is resolved
3. I press space, since I explicitly tell the keyboard that I just want a backtick it restores the word and wrap it in backticks.

After that it shows what happens if I press backtick and then another key that doesn't resolve to a proper char, and finally I press backtick and space, and as you can see it properly creates two backticks.

So I would say it's a won't fix since it works as expected when using the proper keystrokes.

If you don't want the backtick dead key I would recommend using a tool like Ukulele to create a custom layout. Solving that just for Atom probably involve patching Chromium which is not really a smart decision IMO.

**Comment by lee-dohm** [comment]
Thanks @abe33!

@esseb I agree with @abe33 that this should be a won't fix unless there is something that we're missing. I'm going to mark it as such and close the Issue. If there is anything we're missing, please let us know!

Thanks for contributing to Atom @esseb!

**Comment by esseb** [comment]
@abe33 all right, that seems reasonable. Closing the issue. Thanks for investigating.
